components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/filter/CaptchaFilter.java

```diff
@@ -189,4 +190,14 @@ private void doFilter(CaptchaPreValidationResponse preValidationResponse, Servle
             filterChain.doFilter(servletRequest, servletResponse);
         }
     }
+
+    private String extractRedirectURL(HttpServletRequest httpRequest) {
+
+        AuthenticationContext authenticationContext = FrameworkUtils.getContextData(httpRequest);
+        if (authenticationContext != null) {
+            return authenticationContext.getRedirectURL();
+        } else {
+            return httpRequest.getHeader("referer");
```

Comment:
When we are using the referer don't we need the code that attaches the headers as before, otherwise wouldn't this lead to any other issues? 
 https://github.com/wso2-extensions/identity-governance/commit/aaa5695ce2c9896eb52ddac5b0108137f8e5a8fe#diff-fe33aa72259ad318a3b1fdfcffcf1eed48c89830dc2f8e703b53fb18d377d784L177-L180

Reply:
The `referer` URL will contain the header parameters required. On top of that we are adding recaptcha-related header parameters. 
Anyhow the `referer` URL would contain the same endpoint to which the request needs to be redirected in all cases since this is the redirection that happens whenever the captcha is not valid(not ticked) in case it is required. So annexing the recaptcha-related header parameters on top of the referer URL would not cause issues AFAIT (except the spot bug issue)